pkg/controller/storagecluster/exporter.go

```diff
@@ -105,6 +105,7 @@ func CreateOrUpdateService(r *ReconcileStorageCluster, instance *ocsv1.StorageCl
 		return nil, fmt.Errorf("failed to retrieve metrics exporter service %v. %v", namespacedName, err)
 	}
 	service.ResourceVersion = oldService.ResourceVersion
+	service.Spec.ClusterIP = oldService.Spec.ClusterIP
```

Comment:
I am not aware of Updates where we pass old ClusterIP to new Spec.
What issue is this causing?

Reply:
While the operator goes ready, we're flooding the logs with these error messages:

```

2020-10-14T17:32:25.967722321Z {"level":"error","ts":"2020-10-14T17:32:25.967Z","logger":"controller-runtime.controller","msg":"Reconciler error","controller":"storagecluster-controller","request":"openshift-storage/ocs-external-storagecluster","error":"failed to update service openshift-storage/ocs-metrics-exporter. Service \"ocs-metrics-exporter\" is invalid: spec.clusterIP: Invalid value: \"\": field is immutable","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/remote-source/app/vendor/github.com/go-logr/zapr/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/remote-source/app/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:258\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/remote-source/app/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/remote-source/app/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/remote-source/app/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/remote-source/app/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/remote-source/app/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
```

Thus we need to maintain the same value for `ClusterIP`.

I would argue we shouldn't mark Ready if the metrics aren't set up properly, but that's a separate PR.

Reply:
I didn't want to block OCS usage just because we aren't able to set-up exporter (which is currently giving 1 or 2 metrics that we rarely use)